internal/charmstore/store.go

```diff
@@ -335,22 +335,15 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 var everyonePerm = []string{params.Everyone}
 
 func (s *Store) insertEntity(entity *mongodoc.Entity) (err error) {
-	readPerm := everyonePerm
-	var writePerm []string
-	if entity.User != "" {
-		readPerm = []string{params.Everyone, entity.User}
-		writePerm = []string{entity.User}
-	}
 	// Add the base entity to the database.
 	baseEntity := &mongodoc.BaseEntity{
-		URL:  entity.BaseURL,
-		User: entity.User,
-		Name: entity.Name,
-		// TODO frankban: allow specifying non-public charms on initial upload.
-		Public: true,
+		URL:    entity.BaseURL,
+		User:   entity.User,
+		Name:   entity.Name,
+		Public: false,
 		ACLs: mongodoc.ACL{
-			Read:  readPerm,
-			Write: writePerm,
+			Read:  []string{entity.User},
```

Comment:
Could potentially use the same slice for both entries here, saving one ickle bit of garbage.